sklearn/linear_model/_base.py

```diff
@@ -560,8 +560,15 @@ def fit(self, X, y, sample_weight=None):
         y : array-like of shape (n_samples,) or (n_samples, n_targets)
             Target values. Will be cast to X's dtype if necessary.
 
-        sample_weight : array-like of shape (n_samples,), default=None
-            Individual weights for each sample.
+        sample_weight : array-like of shape (n_samples,), int, float or None (default)
+
+            The options are:
+
+            - `array-like`: should contain the individual weight of each sample.
+            - `None`: all samples have a weight equal to 1.
+            - `int` or `float`: all samples have a weight equal to the value \
+provided. Since there is no difference in the relative weight between samples, \
+results in the same fitted model as when `sample_weight=None`.
```

Comment:
I'd go for something concise like
```suggestion
            - `int` or `float`: all samples have a weight equal to the value provided.
```

I wonder if we should explain to the user what exact effect (no effect) this has, we don't go into that for array-like either. So maybe we can skip it.